BraintreeCore/src/main/java/com/braintreepayments/api/ConfigurationLoader.kt

```diff
@@ -0,0 +1,81 @@
+package com.braintreepayments.api
+
+import android.content.Context
+import android.net.Uri
+import android.util.Base64
+import com.braintreepayments.api.Configuration.Companion.fromJson
+import org.json.JSONException
+
+internal class ConfigurationLoader internal constructor(
+    private val httpClient: BraintreeHttpClient,
+    private val configurationCache: ConfigurationCache
+) {
+    constructor(context: Context, httpClient: BraintreeHttpClient) : this(
+        httpClient, ConfigurationCache.getInstance(context)
+    )
+
+    fun loadConfiguration(authorization: Authorization, callback: ConfigurationLoaderCallback) {
+        if (authorization is InvalidAuthorization) {
+            val message = authorization.errorMessage
+            callback.onResult(null, BraintreeException(message))
+            return
+        }
+        val configUrl = Uri.parse(authorization.configUrl)
+            .buildUpon()
+            .appendQueryParameter("configVersion", "3")
+            .build()
+            .toString()
+        val cachedConfig = getCachedConfiguration(authorization, configUrl)
+
+        cachedConfig?.let {
+            callback.onResult(cachedConfig, null)
+        } ?: run {
+            httpClient[configUrl, null, authorization, HttpClient.RETRY_MAX_3_TIMES, HttpResponseCallback { responseBody, httpError ->
```

Comment:
```suggestion
            httpClient.get(configUrl, null, authorization, HttpClient.RETRY_MAX_3_TIMES) { responseBody, httpError ->
```
I'm realizing we're gonna have to rename `HttpClient#get` at some point since it's [reserved in Kotlin](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.js/-json/get.html).